python/taichi/aot/utils.py

```diff
@@ -66,6 +67,20 @@ def produce_injected_args(kernel, symbolic_args=None):
                                   layout=Layout.AOS))
             else:
                 raise RuntimeError('')
+        elif isinstance(anno, MatrixType):
+            if not isinstance(symbolic_args[i], list):
```

Comment:
nit: we can also assert on symbolic_args[i]'s argKind is Matrix here ;)

Reply:
The symbolic args are already expanded here, so there are only Args and (2-D) list of Args here